src/main/scala/model/world/Reproduction.scala

```diff
@@ -24,21 +25,34 @@ object Reproduction {
   /**
    * @param mom a bunny
    * @param dad another bunny
-   * @return the 4 children of the couple, one for each cell of the Punnett's square
+   * @return the 4 children of the couple, one for each cell of the Punnet's square
    */
-  def generateChildren(mom: Bunny, dad: Bunny): Population = {
+  def generateChildren(mom: Bunny, dad: Bunny)(mutations: Option[List[Mutation]]): Population = {
```

Comment:
È solo una questione di gusto ma se vuoi in alternativa al currying puoi mettere il parametro mutations insieme a mom e dad e dargli come default None in questo modo lato chiamante se non si vuole specificare non si deve fare nulla

Reply:
Sono d'accordo con il deafult None così non è da richiamare quando non ci sono nuove mutazioni